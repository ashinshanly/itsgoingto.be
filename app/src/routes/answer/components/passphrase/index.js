import Passphrase from './passphrase'

export default Passphrase
