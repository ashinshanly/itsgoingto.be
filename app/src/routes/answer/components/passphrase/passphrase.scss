@import "variables";

.passphrase-container {
  z-index: 1050;
  position: fixed;
  width: 90%;
  @media (min-width: $screen-sm-min) {
    width: ($container-sm - ($gutter-width * 8));
  }
  @media (min-width: $screen-md-min) {
    width: ($container-md - ($gutter-width * 10));
  }
  @media (min-width: $screen-lg-min) {
    width: ($container-lg - ($gutter-width * 12));
  }
  transform: translate3d(-50%, -50%, 0px);
  top: 30%;
  left: 50%;
}

.input-passphrase {
  margin-bottom: 1em;

  .input-field {
    font-size: 1.5em;
  }
}
