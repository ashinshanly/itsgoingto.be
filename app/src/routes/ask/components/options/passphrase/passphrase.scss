.input-option-passphrase {
  grid-column: 1 / span 1;

  .input-field.input-field-passphrase {
    font-size: 1.2em;
  }
}
