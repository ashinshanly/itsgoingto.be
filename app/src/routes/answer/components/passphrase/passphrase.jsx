import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { withRouter } from 'react-router'
import { mergeAll } from 'ramda'
import { updatePoll } from 'store/poll'
import { fetchPoll, APIError } from 'store/api'
import { setRequiresPassphrase } from 'store/loader'
import EventBus from 'components/event-bus'
import Button from 'components/button'
import './passphrase.scss'

const KEY_ENTER = 13

class Passphrase extends React.Component {
  constructor (props) {
    super(props)
    this.state = {
      error : false
    }
  }

  componentDidMount = () => {
    this.eventBus = EventBus.getEventBus()
  }

  submit = () =>
    this.props.setPassphrase(this.refs.passphrase.value)
      .then(
        () => this.props.fetchPoll(this.props.identifier)
          .then((response) => {
            if (!(response instanceof APIError)) {
              this.props.setRequiresPassphrase(false)
            } else {
              this.setState({
                error : true
              })
            }
          })
      )

  handleKeyPress = (event) => {
    event = event || window.event
    const key = event.keyCode || event.charCode

    if (key === KEY_ENTER) {
      this.eventBus.emit('passphrase-submit')
    }
  }

  render () {
    const { error } = this.state

    return (
      <div className='passphrase-container'>
        <div className={'input-passphrase' + (error ? ' input-error' : '')}>
          <label className='input-label input-label-passphrase' htmlFor='passphrase'>Passphrase</label>
          <input
            className='input-field input-field-passphrase'
            type='text'
            id='passphrase'
            name='passphrase'
            ref='passphrase'
            onKeyDown={this.handleKeyPress} />
          {error &&
            <span className='input-error-label'>
              Passphrase incorrect
            </span>
          }
        </div>
        <Button className='btn pull-right' text='Enter' callback={this.submit} submitEvent='passphrase-submit' />
      </div>
    )
  }
}

Passphrase.propTypes = {
  identifier            : PropTypes.string.isRequired,
  setPassphrase         : PropTypes.func.isRequired,
  fetchPoll             : PropTypes.func.isRequired,
  setRequiresPassphrase : PropTypes.func.isRequired
}

const mapDispatchToProps = (dispatch, props) => ({
  setPassphrase         : (value) => dispatch(updatePoll({ passphrase : value, identifier : props.params.identifier })),
  fetchPoll             : (identifier) => dispatch(fetchPoll(identifier)),
  setRequiresPassphrase : (value) => dispatch(setRequiresPassphrase(value))
})

const mergeProps = (stateProps, dispatchProps, ownProps) => mergeAll([stateProps, dispatchProps, ownProps.params])

export default withRouter(connect(null, mapDispatchToProps, mergeProps)(Passphrase))
