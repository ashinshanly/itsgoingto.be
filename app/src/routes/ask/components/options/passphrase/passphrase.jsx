import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { updatePoll } from 'store/poll/actions'
import './passphrase.scss'

export class Passphrase extends PureComponent {
  handlePassphraseChange = (event) =>
    this.props.updateOptions({
      identifier : '',
      passphrase : event.target.value
    })

  render () {
    const { poll } = this.props

    return (
      <div className='input-option input-option-passphrase'>
        <label
          className='input-label input-label-passphrase'
          htmlFor='passphrase'>
          Set a passphrase (optional)
        </label>
        <input
          className='input-field input-field-passphrase'
          type='text'
          id='passphrase'
          name='passphrase'
          value={poll.passphrase}
          onChange={this.handlePassphraseChange} />
      </div>
    )
  }
}

Passphrase.propTypes = {
  poll          : PropTypes.object.isRequired,
  updateOptions : PropTypes.func.isRequired
}

const mapDispatchToProps = (dispatch) => ({
  updateOptions: (value) => dispatch(updatePoll(value))
})

export default connect(null, mapDispatchToProps)(Passphrase)
