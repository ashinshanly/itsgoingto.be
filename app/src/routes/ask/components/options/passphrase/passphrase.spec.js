/* global expect, jest */
import React from 'react'
import { shallow } from 'enzyme'
import { Passphrase } from './passphrase'

const props = {
  poll          : { passphrase: 'Password' },
  updateOptions : jest.fn(),
}

describe('(Route) Ask', () => {
  describe('(Component) Passphrase', () => {
    describe('(Action) onChange', () => {
      it('should call prop onAnswerChange', () => {
        const wrapper = shallow(<Passphrase {...props} />)
        const event = {
          preventDefault() { },
          target: { value: 'New Password' }
        }

        wrapper.find('input').simulate('change', event)
        expect(props.updateOptions).toHaveBeenCalledWith({
          identifier : '',
          passphrase : 'New Password',
        })
      })
    })

    describe('(Render)', () => {
      it('matches snapshot', () => {
        const wrapper = shallow(<Passphrase {...props} />)
        expect(wrapper).toMatchSnapshot()
      })
    })
  })
})
