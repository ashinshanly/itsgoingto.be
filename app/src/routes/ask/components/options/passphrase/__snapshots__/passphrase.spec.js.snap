// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Route) Ask (Component) Passphrase (Render) matches snapshot 1`] = `
<div
  className="input-option input-option-passphrase"
>
  <label
    className="input-label input-label-passphrase"
    htmlFor="passphrase"
  >
    Set a passphrase (optional)
  </label>
  <input
    className="input-field input-field-passphrase"
    id="passphrase"
    name="passphrase"
    onChange={[Function]}
    type="text"
    value="Password"
  />
</div>
`;
